src/pages/resultsView/survival/SurvivalChart.tsx

```diff
@@ -594,6 +598,51 @@ export default class SurvivalChart
         this.sliderValue = Number.parseFloat(text);
     }
 
+    @autobind
+    private onSelection(data: any[]) {
+        const scatterPoints: Array<ScatterData> = data[1].data;
+        if (scatterPoints.length > 2) {
+            const dataBin = this.generateFilterDataBin(scatterPoints);
+            if (this.props.onUserSelection) {
+                this.props.onUserSelection([dataBin]);
+            }
+        }
+    }
+
+    private generateFilterDataBin(scatterPoints: Array<ScatterData>): DataBin {
```

Comment:
just wanted to point out that it would probably be better to make this a pure helper function rather than a private class method.   Were you to write a unit test for this, it would be much easier to do so.  It relies on that id prop, but that should be a function argument so that it's role as input is made clear.